src/ads/MobileAd.ts

```diff
@@ -76,19 +88,44 @@ export class MobileAd {
       this._isLoadCalled = false;
     }
 
-    if (this._onAdEventHandler) {
-      let nativeError;
-      if (error) {
-        nativeError = NativeError.fromEvent(error, 'googleMobileAds');
-      }
+    let payload: AdEventPayload<EventType> = data;
+    if (error) {
+      payload = NativeError.fromEvent(error, 'googleMobileAds');
+    }
+    this._adEventsListeners.forEach(listener => {
+      listener({
+        type,
+        payload,
+      });
+    });
+    this._getAdEventListeners(type).forEach(listener => {
+      listener(payload);
+    });
+  }
+
+  protected _addAdEventsListener(listener: AdEventsListener) {
+    const index = this._adEventsListeners.push(listener as AdEventsListener<EventType>) - 1;
+    return () => {
+      this._adEventsListeners.splice(index, 1);
```

Comment:
Oh, I didn't think of this.

Reply:
I only thought of it because this is a specific dragon that I have fought before, and it was really hard. In that case I *had* to maintain ordering, so I could not change data structure, and I had to implement a "change list" that could "render" things with the correct ordinal movement as adds and deletes happened, and then "compact" things correctly. It was terrible

I think using a different data structure that does not care about the ordinals (array indexes) is the easiest way for sure

Reply:
I changed to Map structure finally.